Analysis/LeptonID/src/PIDutil.cc

```diff
@@ -0,0 +1,374 @@
+#include "PIDutil.h"
+#include <cmath>
+#include <set>
+#include <TF2.h>
+
+#include <marlin/VerbosityLevels.h>
+
+#include <EVENT/ReconstructedParticle.h>
+
+float getCorrectdEdx(float dEdx, Track *track)
+{
+    // dEdx correction stolen from the dEdx correction processor
+    double lambda = abs(atan(track->getTanLambda()) * 180.0 / M_PI);
+    std::vector<double> par = {0.970205, 0.0007506, 4.41781e-8, 5.8222e-8};
+
+    double f3 = 1 / (par[0] + par[1] * lambda + par[2] * pow(lambda, 2) + par[3] * pow(lambda, 3));
+
+    float dEdx_v2 = dEdx * f3;
+    return dEdx_v2;
+}
+
+LCObject *getRelated(LCObject *p, LCRelationNavigator const &nav, int &_cw, int &_tw)
```

Comment:
If I understand this correctly, this function could almost entirely be replaced by new functionality in the [`LCRelationNavigator`](https://github.com/iLCSoft/LCIO/blob/ed1a20027baf614d53758c8dd279411749938267/src/cpp/include/UTIL/LCRelationNavigator.h#L85-L120) and [`MarlinUtil`](https://github.com/iLCSoft/MarlinUtil/blob/402540c525b98688e098ddbf2f3957eb9148d2f0/source/include/MarlinUtil.h#L64-L80)

Reply:
Good idea, these weren't there when I wrote this.

This would be straightforward if [`getMaxWeightIdx`](https://github.com/iLCSoft/LCIO/blob/ed1a20027baf614d53758c8dd279411749938267/src/cpp/include/UTIL/LCRelationNavigator.h#L27-L35) was `public`. Even then, implementing the same logic would call it twice to loop over the weights. If I use `getRelatedToMaxWeightObject` and `getRelatedToMaxWeight` to emulate the same logic we will loop over the weights four times. Not sure if this will have any meaningful impact but for me it feels wrong to do that.

tl; dr: I would prefer to have a `getRelatedToMaxWeightIndex` function in `LCRelationNavigator` to do this.. Can we still get that in even though you just did a release @tmadlener?

Reply:
`getMaxWeightIdx` is a free function in the `UTIL` namespace, it should be possible to call that just passing in the corresponding weights, [similar to what is done in the LCRelationNavigator](https://github.com/iLCSoft/LCIO/blob/ed1a20027baf614d53758c8dd279411749938267/src/cpp/include/UTIL/LCRelationNavigator.h#L117-L119).

I do agree that having something that returns the object and the weight would be useful in your case. No problem adding that (even if we just had a new tag).

Reply:
Thx, must have overlooked that. I am using `getMaxWeightIdx` now. I can add a method that returns object and weight to the LCRelationNavigator later and update PIDUtil in a future PR. I am happy enough with this for now :)